File: SudokuCollective.WebApi/client/src/models/viewModels/pageListModel.js
import { SortValue } from "../viewModels/sortValue";

class PageListModel {

    constructor(page, itemsPerPage, sortBy, orderByDescending, includeCompletedGames) {

        if (!arguments.length) {

            this.page = 0;
            this.itemsPerPage = 0;
            this.sortBy = SortValue[0].value;
            this.orderByDescending = false;
            this.includeCompletedGames = false;

        } else {

            this.page = page;
            this.itemsPerPage = itemsPerPage;
            this.sortBy = sortBy;
            this.orderByDescending = orderByDescending;
            this.includeCompletedGames = includeCompletedGames;
        }
    }
}

export default PageListModel;

File: SudokuCollective.Data/Models/PageModels/PageListModel.cs
using SudokuCollective.Core.Enums;
using SudokuCollective.Core.Interfaces.APIModels.PageModels;

namespace SudokuCollective.Data.Models.PageModels
{
    public class PageListModel : IPageListModel
    {
        public int Page { get; set; }
        public int ItemsPerPage { get; set; }
        public SortValue SortBy { get; set; }
        public bool OrderByDescending { get; set; }
        public bool IncludeCompletedGames { get; set; }

        public PageListModel()
        {
            Page = 0;
            ItemsPerPage = 0;
            SortBy = SortValue.NULL;
            OrderByDescending = false;
            IncludeCompletedGames = false;
        }

        public PageListModel(
            int page,
            int itemsPerPage,
            int sortValue,
            bool orderByDescending,
            bool includeCompletedGames)
        {
            Page = page;
            ItemsPerPage = itemsPerPage;
            SortBy = (SortValue)sortValue;
            OrderByDescending = orderByDescending;
            IncludeCompletedGames = includeCompletedGames;
        }

        public bool IsNull()
        {
            var result = false;

            if (Page == 0
                && ItemsPerPage == 0
                && SortBy == SortValue.NULL
                && OrderByDescending == false
                && IncludeCompletedGames == false)
            {
                result = true;
            }

            return result;
        }
    }
}
